tests/track/params_test.py

```diff
@@ -151,6 +151,19 @@ def conflict(action, id):
         # and we're back to random
         self.assertEqual(conflict(conflict_action, "100"), next(generator))
 
+    def test_generate_action_meta_data_with_id_and_zero_conflict_probability(self):
+        def idx(id):
+            return "index", '{"index": {"_index": "test_index", "_type": "test_type", "_id": "%s"}}' % id
+
+        generator = params.GenerateActionMetaData("test_index", "test_type",
+                                                  conflicting_ids=[100, 200, 300, 400],
+                                                  conflict_probability=0)
+
+        self.assertEqual(idx("100"), next(generator))
+        self.assertEqual(idx("200"), next(generator))
+        self.assertEqual(idx("300"), next(generator))
+        self.assertEqual(idx("400"), next(generator))
```

Comment:
That leaves the possibility that the generator is creating more entries than you expect. You could convert it to a list and compare the list contents to be sure it is only generating four elements?